File: camera_F200.cpp
#include <librealsense/rs.hpp> 
#include <opencv2/opencv.hpp>

int main() {
    rs2::pipeline pipe;
    rs2::config cfg;
    cfg.enable_stream(RS2_STREAM_COLOR, 640, 480, RS2_FORMAT_BGR8, 30);

    pipe.start(cfg);

    cv::namedWindow("RealSense Display", cv::WINDOW_AUTOSIZE);

    while (true) {
        rs2::frameset frames = pipe.wait_for_frames();

        rs2::frame color_frame = frames.get_color_frame();
        const int w = color_frame.as<rs2::video_frame>().get_width();
        const int h = color_frame.as<rs2::video_frame>().get_height();

        cv::Mat image(cv::Size(w, h), CV_8UC3, (void*)color_frame.get_data(), cv::Mat::AUTO_STEP);

        cv::imshow("RealSense Display", image);

        if (cv::waitKey(1) == 'q') {
            break;
        }
    }

    pipe.stop();
    cv::destroyAllWindows();

    return 0;
}

File: RGBD_Camera_intelRealsense/camera_F200.py
import cv2
import numpy as np
import pyrealsense2 as rs

# Configurando el pipeline
pipeline = rs.pipeline()
config = rs.config()
config.enable_stream(rs.stream.color, 640, 480, rs.format.bgr8, 30)

# Iniciando el pipeline
pipeline.start(config)

try:
    while True:
        # Esperando un nuevo frame
        frames = pipeline.wait_for_frames()

        # Obteniendo el frame de color
        color_frame = frames.get_color_frame()
        if not color_frame:
            continue

        # Convertir el frame a una imagen OpenCV
        color_image = np.asanyarray(color_frame.get_data())

        # Mostrar la imagen
        cv2.imshow('RealSense SR300', color_image)

        # Salir al presionar la tecla 'q'
        if cv2.waitKey(1) & 0xFF == ord('q'):
            break

finally:
    # Cerrar el pipeline
    pipeline.stop()
    cv2.destroyAllWindows()

